src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/adjudications/AdjudicationService.kt

```diff
@@ -614,28 +614,53 @@ class AdjudicationService(
   }
 
   @Audit
-  fun createHearingResult(
+  fun upsertHearingResult(
     adjudicationNumber: Long,
     hearingId: Long,
     chargeSequence: Int,
     request: CreateHearingResultRequest,
   ) {
-    if (validateFindingCode(request.findingCode)) {
-      // DPS only allows 1 result per hearing, the created result has a result sequence of 1
-      val party = adjudicationIncidentPartyRepository.findByAdjudicationNumber(adjudicationNumber)
-        ?: throw NotFoundException("Adjudication party with adjudication number $adjudicationNumber not found")
+    val telemetryMap = mutableMapOf(
+      "adjudicationNumber" to adjudicationNumber.toString(),
+      "chargeSequence" to chargeSequence.toString(),
+      "hearingId" to hearingId.toString(),
+      "findingCode" to request.findingCode,
+      "plea" to request.pleaFindingCode,
+    )
+
+    // DPS only allows 1 result per hearing, the created result has a result sequence of 1
+    val party = adjudicationIncidentPartyRepository.findByAdjudicationNumber(adjudicationNumber)
+      ?: throw NotFoundException("Adjudication party with adjudication number $adjudicationNumber not found")
+
+    // DPS created (or migrated) adjudications will have 1 charge
+    val incidentCharge = party.charges.firstOrNull { it.id.chargeSequence == chargeSequence }
+      ?: throw NotFoundException("Charge not found for adjudication number $adjudicationNumber and charge sequence $chargeSequence")
 
-      // DPS created (or migrated) adjudications will have 1 charge
-      val incidentCharge = party.charges.firstOrNull { it.id.chargeSequence == chargeSequence }
-        ?: throw NotFoundException("Charge not found for adjudication number $adjudicationNumber and charge sequence $chargeSequence")
+    val hearing = adjudicationHearingRepository.findByIdOrNull(hearingId)
+      ?: throw NotFoundException("Hearing not found. Hearing Id: $hearingId")
 
-      val hearing = adjudicationHearingRepository.findByIdOrNull(hearingId)
-        ?: throw NotFoundException("Hearing not found. Hearing Id: $hearingId")
+    request.adjudicatorUsername?.let { hearing.hearingStaff = findStaffByUsername(request.adjudicatorUsername) }
 
-      request.adjudicatorUsername?.let { hearing.hearingStaff = findStaffByUsername(request.adjudicatorUsername) }
+    val existingResult = adjudicationHearingResultRepository.findFirstOrNullById_OicHearingIdAndChargeSequence(
+      chargeSequence = chargeSequence,
+      hearingId = hearingId,
+    )
 
+    existingResult?.let {
+      it.pleaFindingType = lookupPleaFindingType(request.pleaFindingCode)
+      it.pleaFindingCode = request.pleaFindingCode
+      it.findingType = lookupFindingType(request.findingCode)
+      adjudicationHearingResultRepository.saveAndFlush(it)
```

Comment:
save and flushing here to stop an app insights created message when DB write could fail. Maybe we don't usually bother but trying to avoid confusion on error